lib/spack/env/cc

```diff
@@ -272,7 +272,7 @@ for dep in "${deps[@]}"; do
     # Prepend include directories
     if [[ -d $dep/include ]]; then
         if [[ $mode == cpp || $mode == cc || $mode == as || $mode == ccld ]]; then
-            args=("-I$dep/include" "${args[@]}")
```

Comment:
These are not intentional, probably? If so, please undo/rebase.

Reply:
Same above

Reply:
Yes you're right they are not intended to stay, they just highlight a build issue I ran into.  See the PR description paragraph re: the "workaround" to get VTK to build.  Perhaps @mathstuf knows how to manage this from the VTK side.  But because the spack-installed package headers are *prepended* to the include list, those includes are found before VTK's ThirdParty includes, which VTK needs to use.  In this particular case (libtiff), even though the spack-installed libtiff headers were found (indicating jbig support is available), the library with the jbig symbols wasn't found at link time, so the build failed.  See the commit [message](https://github.com/spack/spack/pull/6551/commits/4da36c8fdbd04e4d97067499e3b6c5ff4d47a388) of that workaround change for some more detail.

Reply:
If Spack is providing the dependencies, VTK should be told to build using those, not its internal copies.

Reply:
I see, thanks.  So do you have any idea why VTK would find the spack libtiff includes, but not the associated libs?

Reply:
That's something that `CMakeCache.txt` is probably better suited to answer :) .